### ISSUE TITLE
CSI pods in CrashLoopBackOff when one GUI pod is in Terminating state, Node down scenario 

### ISSUE THREAD

**Issue by rkomandu** [IBM/ibm-spectrum-scale-csi#634]
## Describe the bug

When the worker node is made down on a 3M+3W (Fyre) configuration, the CSI pods are getting into CrashLoopBackoffState and Noobaa-db-pg-0 is unable to migrate to the other worker node where the CSI attacher is running and another GUI pod is Active. CSI pods should communicate and allow the noobaa-db to migrate with the respective PV, but that doesn't seem to be the case. 

**Note: CNSA defect 2951 is opened** 

## How to Reproduce?
Please list the steps to help development teams reproduce the behavior

I am trying to recreate a Noobaa defect 6869. It has a pod by name noobaa-db-pg-0 and has a PV attached to it. 
Please see steps mentioned below 

```
Step1: Noobaa db-pg pod is running on worker1

Every 3.0s: oc get pods -n openshift-storage -o wide; oc get pods -n ibm-spectrum-scale-das -o wide          api.rkomandu-ta.cp.fyre.ibm.com: Mon Feb 21 22:56:00 2022

NAME                                               READY   STATUS    RESTARTS       AGE     IP              NODE                                  NOMINATED NODE   REA
DINESS GATES
noobaa-core-0                                      1/1     Running   0              83m     10.254.16.199   worker0.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
noobaa-db-pg-0                                     1/1     Running   0              79m     10.254.13.24    worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
noobaa-default-backing-store-noobaa-pod-77176233   1/1     Running   0              90m     10.254.13.22    worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
noobaa-endpoint-74b7459c5f-2xszg                   1/1     Running   0              2m18s   10.254.20.7     worker2.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
noobaa-endpoint-74b7459c5f-gxl67                   1/1     Running   0              91m     10.254.16.194   worker0.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
noobaa-endpoint-74b7459c5f-r7bh8                   1/1     Running   0              79m     10.254.13.23    worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
noobaa-operator-54877b7dc9-nvms5                   1/1     Running   0              24h     10.254.13.4     worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
ocs-metrics-exporter-7955bfc785-mtl76              1/1     Running   0              24h     10.254.13.1     worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
ocs-operator-57d785c8c7-lnjx2                      1/1     Running   2 (162m ago)   24h     10.254.13.3     worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
odf-console-756c9c8bc7-s2vfs                       1/1     Running   0              24h     10.254.12.255   worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
odf-operator-controller-manager-89746b599-zjn6f    2/2     Running   2 (162m ago)   24h     10.254.13.0     worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
rook-ceph-operator-74864f7c6f-9xt5x                1/1     Running   0              5d23h   10.254.12.246   worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>


Step 2: CNSA , CSI pods in their respective NS (Here GUI is running on worker1, worker0.  CSI Attacher on worker0 

Every 5.0s: oc get pods -n ibm-spectrum-scale -o wide ; oc get pods -n ibm-spectrum-scale-csi -o wide        api.rkomandu-ta.cp.fyre.ibm.com: Mon Feb 21 22:56:16 2022

NAME                                                  READY   STATUS    RESTARTS   AGE     IP              NODE                                  NOMINATED NODE   READ
INESS GATES
ibm-spectrum-scale-gui-0                              4/4     Running   0          12d     10.254.12.238   worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <non
e>
ibm-spectrum-scale-gui-1                              4/4     Running   0          23h     10.254.16.192   worker0.rkomandu-ta.cp.fyre.ibm.com   <none>           <non
e>
ibm-spectrum-scale-noobaamonitoring-bdcb4cbbb-4mpn4   1/1     Running   0          24h     10.254.13.9     worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <non
e>
ibm-spectrum-scale-pmcollector-0                      2/2     Running   0          12d     10.254.12.239   worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <non
e>
ibm-spectrum-scale-pmcollector-1                      2/2     Running   0          23h     10.254.16.193   worker0.rkomandu-ta.cp.fyre.ibm.com   <none>           <non
e>
worker0                                               2/2     Running   2          5d18h   10.17.126.141   worker0.rkomandu-ta.cp.fyre.ibm.com   <none>           <non
e>
worker1                                               2/2     Running   0          2d11h   10.17.126.253   worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <non
e>
worker2                                               2/2     Running   12         21d     10.17.127.141   worker2.rkomandu-ta.cp.fyre.ibm.com   <none>           <non
e>
NAME                                              READY   STATUS    RESTARTS        AGE     IP              NODE                                  NOMINATED NODE   REA
DINESS GATES
ibm-spectrum-scale-csi-attacher-0                 1/1     Running   21              74d     10.254.16.5     worker0.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
ibm-spectrum-scale-csi-dsgj5                      3/3     Running   33              5d17h   10.17.127.141   worker2.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
ibm-spectrum-scale-csi-mg9cj                      3/3     Running   3               5d17h   10.17.126.141   worker0.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
ibm-spectrum-scale-csi-operator-d844fb754-s4lst   1/1     Running   1 (7h56m ago)   24h     10.254.13.8     worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
ibm-spectrum-scale-csi-provisioner-0              1/1     Running   21              74d     10.254.16.3     worker0.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
ibm-spectrum-scale-csi-resizer-0                  1/1     Running   21              74d     10.254.16.2     worker0.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
ibm-spectrum-scale-csi-snapshotter-0              1/1     Running   21              74d     10.254.16.4     worker0.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>
ibm-spectrum-scale-csi-z5gcm                      3/3     Running   0               2d11h   10.17.126.253   worker1.rkomandu-ta.cp.fyre.ibm.com   <none>           <no
ne>

Step 3: Node status 

oc get nodes
NAME                                  STATUS   ROLES    AGE   VERSION
master0.rkomandu-ta.cp.fyre.ibm.com   Ready    master   74d   v1.22.0-rc.0+a44d0f0
master1.rkomandu-ta.cp.fyre.ibm.com   Ready    master   74d   v1.22.0-rc.0+a44d0f0
master2.rkomandu-ta.cp.fyre.ibm.com   Ready    master   74d   v1.22.0-rc.0+a44d0f0
worker0.rkomandu-ta.cp.fyre.ibm.com   Ready    worker   74d   v1.22.0-rc.0+a44d0f0
worker1.rkomandu-ta.cp.fyre.ibm.com   Ready    worker   74d   v1.22.0-rc.0+a44d0f0
worker2.rkomandu-ta.cp.fyre.ibm.com   Ready    worker   74d   v1.22.0-rc.0+a44d0f0

Step 4: make worker1 down via Fyre UI 

root@api.rkomandu-ta.cp.fyre.ibm.com ODF-4.9.2]# oc get nodes
NAME                                  STATUS     ROLES    AGE   VERSION
master0.rkomandu-ta.cp.fyre.ibm.com   Ready      master   74d   v1.22.0-rc.0+a44d0f0
master1.rkomandu-ta.cp.fyre.ibm.com   Ready      master   74d   v1.22.0-rc.0+a44d0f0
master2.rkomandu-ta.cp.fyre.ibm.com   Ready      master   74d   v1.22.0-rc.0+a44d0f0
worker0.rkomandu-ta.cp.fyre.ibm.com   Ready      worker   74d   v1.22.0-rc.0+a44d0f0
**worker1.rkomandu-ta.cp.fyre.ibm.com   NotReady   worker   74d   v1.22.0-rc.0+a44d0f0**
worker2.rkomandu-ta.cp.fyre.ibm.com   Ready      worker   74d   v1.22.0-rc.0+a44d0f0


Step 5: Noobaa-db-pg-0 is not migrated, **it is continuous to be in Init state**
Every 3.0s: oc get pods -n openshift-storage -o wide; oc get pods -n ibm-spectrum-scale-das -o wide          api.rkomandu-ta.cp.fyre.ibm.com: Mon Feb 21 23:05:33 2022

NAME                                              READY   STATUS        RESTARTS       AGE     IP              NODE                                  NOMINATED NODE
READINESS GATES
noobaa-core-0                                     1/1     Running       0              93m     10.254.16.199   worker0.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
noobaa-db-pg-0                                    0/1     Init:0/2      0              2m45s   <none>          worker2.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
noobaa-endpoint-74b7459c5f-2xszg                  1/1     Running       0              11m     10.254.20.7     worker2.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
noobaa-endpoint-74b7459c5f-gxl67                  1/1     Running       0              100m    10.254.16.194   worker0.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
noobaa-endpoint-74b7459c5f-k4t2l                  0/1     Pending       0              2m53s   <none>          <none>                                <none>
<none>
noobaa-operator-54877b7dc9-dss67                  1/1     Running       0              2m53s   10.254.20.24    worker2.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
ocs-metrics-exporter-7955bfc785-96ccq             1/1     Running       0              2m53s   10.254.20.17    worker2.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
ocs-metrics-exporter-7955bfc785-mtl76             1/1     Terminating   0              24h     10.254.13.1     worker1.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
ocs-operator-57d785c8c7-6nhw5                     1/1     Running       0              2m53s   10.254.20.26    worker2.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
ocs-operator-57d785c8c7-lnjx2                     1/1     Terminating   2 (172m ago)   24h     10.254.13.3     worker1.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
odf-console-756c9c8bc7-8h6pt                      1/1     Running       0              2m53s   10.254.20.16    worker2.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
odf-console-756c9c8bc7-s2vfs                      1/1     Terminating   0              24h     10.254.12.255   worker1.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
odf-operator-controller-manager-89746b599-hl7k9   2/2     Running       0              2m54s   10.254.20.15    worker2.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
odf-operator-controller-manager-89746b599-zjn6f   2/2     Terminating   2 (172m ago)   24h     10.254.13.0     worker1.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
rook-ceph-operator-74864f7c6f-9xt5x               1/1     Terminating   0              5d23h   10.254.12.246   worker1.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
rook-ceph-operator-74864f7c6f-q5xd7               1/1     Running       0              2m53s   10.254.16.213   worker0.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>

noobaa-db-pg-0                                    0/1     Init:0/2      0               178m    <none>          worker2.rkomandu-ta.cp.fyre.ibm.com   <none>
 <none>


Step 6: CNSA, CSI pods after the worker1 is down (GUI-0 is Terminating, CSI pods on worker2,worker0 are in CrashLoopBackOff)

Every 5.0s: oc get pods -n ibm-spectrum-scale -o wide ; oc get pods -n ibm-spectrum-scale-csi -o wide        api.rkomandu-ta.cp.fyre.ibm.com: Mon Feb 21 23:05:56 2022

NAME                                                  READY   STATUS        RESTARTS   AGE     IP              NODE                                  NOMINATED NODE
READINESS GATES
ibm-spectrum-scale-gui-0                              4/4     Terminating   0          12d     10.254.12.238   worker1.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
ibm-spectrum-scale-gui-1                              4/4     Running       0          23h     10.254.16.192   worker0.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
ibm-spectrum-scale-noobaamonitoring-bdcb4cbbb-4mpn4   1/1     Terminating   0          24h     10.254.13.9     worker1.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
ibm-spectrum-scale-noobaamonitoring-bdcb4cbbb-7blb5   1/1     Running       0          3m17s   10.254.20.14    worker2.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
ibm-spectrum-scale-pmcollector-0                      2/2     Terminating   0          12d     10.254.12.239   worker1.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
ibm-spectrum-scale-pmcollector-1                      2/2     Running       0          23h     10.254.16.193   worker0.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
worker0                                               2/2     Running       2          5d18h   10.17.126.141   worker0.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
worker1                                               2/2     Running       0          2d11h   10.17.126.253   worker1.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>
worker2                                               2/2     Running       12         21d     10.17.127.141   worker2.rkomandu-ta.cp.fyre.ibm.com   <none>
<none>

NAME                                              READY   STATUS             RESTARTS         AGE     IP              NODE                                  NOMINATED
NODE   READINESS GATES
ibm-spectrum-scale-csi-attacher-0                 1/1     Running            22 (4m25s ago)   74d     10.254.16.5     worker0.rkomandu-ta.cp.fyre.ibm.com   <none>
       <none>
ibm-spectrum-scale-csi-dsgj5                      2/3     CrashLoopBackOff   39 (57s ago)     5d17h   10.17.127.141   worker2.rkomandu-ta.cp.fyre.ibm.com   <none>
       <none>
ibm-spectrum-scale-csi-mg9cj                      2/3     CrashLoopBackOff   9 (76s ago)      5d17h   10.17.126.141   worker0.rkomandu-ta.cp.fyre.ibm.com   <none>
       <none>
ibm-spectrum-scale-csi-operator-d844fb754-s4lst   1/1     Terminating        1 (8h ago)       24h     10.254.13.8     worker1.rkomandu-ta.cp.fyre.ibm.com   <none>
       <none>
ibm-spectrum-scale-csi-operator-d844fb754-w7mml   1/1     Running            0                3m16s   10.254.20.19    worker2.rkomandu-ta.cp.fyre.ibm.com   <none>
       <none>
ibm-spectrum-scale-csi-provisioner-0              1/1     Running            22 (4m25s ago)   74d     10.254.16.3     worker0.rkomandu-ta.cp.fyre.ibm.com   <none>
       <none>
ibm-spectrum-scale-csi-resizer-0                  1/1     Running            22 (4m25s ago)   74d     10.254.16.2     worker0.rkomandu-ta.cp.fyre.ibm.com   <none>
       <none>
ibm-spectrum-scale-csi-snapshotter-0              1/1     Running            22 (4m25s ago)   74d     10.254.16.4     worker0.rkomandu-ta.cp.fyre.ibm.com   <none>
       <none>
ibm-spectrum-scale-csi-z5gcm                      3/3     Running            0                2d11h   10.17.126.253   worker1.rkomandu-ta.cp.fyre.ibm.com   <none>
       <none>

Step 7: noobaa-db-pg-0 describe pod 

 oc describe pod noobaa-db-pg-0 -n openshift-storage
Name:           noobaa-db-pg-0
Namespace:      openshift-storage
Priority:       0
Node:           worker2.rkomandu-ta.cp.fyre.ibm.com/10.17.127.141
Start Time:     Mon, 21 Feb 2022 23:03:29 -0800
Labels:         app=noobaa
                controller-revision-hash=noobaa-db-pg-78657cd7c4
                noobaa-db=postgres
                statefulset.kubernetes.io/pod-name=noobaa-db-pg-0
Annotations:    openshift.io/scc: noobaa-endpoint
Status:         Pending
IP:
IPs:            <none>
Controlled By:  StatefulSet/noobaa-db-pg
Init Containers:
  init:
    Container ID:
    Image:         quay.io/rhceph-dev/odf4-mcg-core-rhel8@sha256:5507f2c1074bfb023415f0fef16ec42fbe6e90c540fc45f1111c8c929e477910
...

Events:
  Type     Reason              Age                  From                     Message
  ----     ------              ----                 ----                     -------
  Normal   Scheduled           74m                  default-scheduler        Successfully assigned openshift-storage/noobaa-db-pg-0 to worker2.rkomandu-ta.cp.fyre.ibm.com
  Warning  FailedAttachVolume  74m                  attachdetach-controller  Multi-Attach error for volume "pvc-a50e49cb-ef07-45fc-ad4e-03d5530db442" Volume is already exclusively attached to one node and can't be attached to another
  Warning  FailedMount         9m11s (x5 over 72m)  kubelet                  Unable to attach or mount volumes: unmounted volumes=[db], unattached volumes=[noobaa-postgres-initdb-sh-volume noobaa-postgres-config-volume db kube-api-access-prf46]: timed out waiting for the condition
  Warning  FailedMount         4m37s (x5 over 38m)  kubelet                  Unable to attach or mount volumes: unmounted volumes=[db], unattached volumes=[noobaa-postgres-config-volume db kube-api-access-prf46 noobaa-postgres-initdb-sh-volume]: timed out waiting for the condition
  Warning  FailedMount         2m23s (x3 over 68m)  kubelet                  Unable to attach or mount volumes: unmounted volumes=[db], unattached volumes=[kube-api-access-prf46 noobaa-postgres-initdb-sh-volume noobaa-postgres-config-volume db]: timed out waiting for the condition
  Warning  FailedMount         5s (x20 over 65m)    kubelet                  Unable to attach or mount volumes: unmounted volumes=[db], unattached volumes=[db kube-api-access-prf46 noobaa-postgres-initdb-sh-volume noobaa-postgres-config-volume]: timed out waiting for the condition


Step 8: 
oc describe pod ibm-spectrum-scale-csi-dsgj5 -n ibm-spectrum-scale-csi
Name:         ibm-spectrum-scale-csi-dsgj5
Namespace:    ibm-spectrum-scale-csi
Priority:     0
Node:         worker2.rkomandu-ta.cp.fyre.ibm.com/10.17.127.141
Start Time:   Wed, 16 Feb 2022 05:34:45 -0800
Labels:       app=ibm-spectrum-scale-csi
              app.kubernetes.io/instance=ibm-spectrum-scale-csi-operator
              app.kubernetes.io/managed-by=ibm-spectrum-scale-csi-operator
              app.kubernetes.io/name=ibm-spectrum-scale-csi-operator
              controller-revision-hash=6d4dcdb4f
              pod-template-generation=3
              product=ibm-spectrum-scale-csi
              release=ibm-spectrum-scale-csi-operator
Annotations:  openshift.io/scc: spectrum-scale-csiaccess
              productID: ibm-spectrum-scale-csi-operator
              productName: IBM Spectrum Scale CSI Operator
              productVersion: 2.4.0
...


Step 9: CSI pod on worker2
oc logs ibm-spectrum-scale-csi-dsgj5 -c ibm-spectrum-scale-csi -n ibm-spectrum-scale-csi
I0222 08:21:03.632548       1 gpfs.go:81] gpfs GetScaleDriver
I0222 08:21:03.632610       1 gpfs.go:157] gpfs SetupScaleDriver. name: spectrumscale.csi.ibm.com, version: 2.4.0, nodeID: worker2.rkomandu-ta.cp.fyre.ibm.com
I0222 08:21:03.632617       1 gpfs.go:197] gpfs PluginInitialize
I0222 08:21:03.632621       1 scale_config.go:93] scale_config LoadScaleConfigSettings
I0222 08:21:03.632791       1 scale_config.go:116] scale_config HandleSecrets
I0222 08:21:03.632866       1 gpfs.go:397] gpfs ValidateScaleConfigParameters.
I0222 08:21:03.632884       1 connectors.go:104] connector GetSpectrumScaleConnector
I0222 08:21:03.632891       1 rest_v2.go:120] rest_v2 NewSpectrumRestV2.
I0222 08:21:03.632897       1 rest_v2.go:142] Created Spectrum Scale connector without SSL mode for ibm-spectrum-scale-gui.ibm-spectrum-scale
I0222 08:21:03.632900       1 rest_v2.go:157] rest_v2 GetClusterId
I0222 08:21:03.632907       1 http_utils.go:60] http_utils FormatURL. url: https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/
I0222 08:21:03.632915       1 rest_v2.go:927] rest_v2 doHTTP. endpoint: https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgmt/v2/cluster, method: GE        T, param: <nil>
I0222 08:21:03.632920       1 http_utils.go:74] http_utils HttpExecuteUserAuth. type: GET, url: https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgm        t/v2/cluster, user: CsiAdmin
E0222 08:21:06.684875       1 rest_v2.go:931] Error in authentication request: Get "https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgmt/v2/cluster        ": dial tcp 172.30.136.191:443: connect: no route to host
E0222 08:21:06.684946       1 rest_v2.go:164] Unable to get cluster ID: Get "https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgmt/v2/cluster": dial         tcp 172.30.136.191:443: connect: no route to host
E0222 08:21:06.684978       1 gpfs.go:222] Error getting cluster ID: Get "https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgmt/v2/cluster": dial tc        p 172.30.136.191:443: connect: no route to host
E0222 08:21:06.684991       1 gpfs.go:164] Error in plugin initialization: Get "https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgmt/v2/cluster": d        ial tcp 172.30.136.191:443: connect: no route to host
F0222 08:21:06.685003       1 main.go:70] Failed to initialize Scale CSI Driver: Get "https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgmt/v2/clust        er": dial tcp 172.30.136.191:443: connect: no route to host

Step 10: another CSI pod in crashLoopBackOff state on worker0

oc logs ibm-spectrum-scale-csi-mg9cj -c ibm-spectrum-scale-csi -n ibm-spectrum-scale-csi
I0222 10:04:24.975688       1 gpfs.go:81] gpfs GetScaleDriver
I0222 10:04:24.975779       1 gpfs.go:157] gpfs SetupScaleDriver. name: spectrumscale.csi.ibm.com, version: 2.4.0, nodeID: worker0.rkomandu-ta.cp.fyre.ibm.com
I0222 10:04:24.975786       1 gpfs.go:197] gpfs PluginInitialize
I0222 10:04:24.975790       1 scale_config.go:93] scale_config LoadScaleConfigSettings
I0222 10:04:24.975949       1 scale_config.go:116] scale_config HandleSecrets
I0222 10:04:24.976022       1 gpfs.go:397] gpfs ValidateScaleConfigParameters.
I0222 10:04:24.976036       1 connectors.go:104] connector GetSpectrumScaleConnector
I0222 10:04:24.976041       1 rest_v2.go:120] rest_v2 NewSpectrumRestV2.
I0222 10:04:24.976047       1 rest_v2.go:142] Created Spectrum Scale connector without SSL mode for ibm-spectrum-scale-gui.ibm-spectrum-scale
I0222 10:04:24.976052       1 rest_v2.go:157] rest_v2 GetClusterId
I0222 10:04:24.976057       1 http_utils.go:60] http_utils FormatURL. url: https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/
I0222 10:04:24.976065       1 rest_v2.go:927] rest_v2 doHTTP. endpoint: https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgmt/v2/cluster, method: GET, param: <nil>
I0222 10:04:24.976072       1 http_utils.go:74] http_utils HttpExecuteUserAuth. type: GET, url: https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgmt/v2/cluster, user: CsiAdmin
E0222 10:04:28.041072       1 rest_v2.go:931] Error in authentication request: Get "https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgmt/v2/cluster": dial tcp 172.30.136.191:443: connect: no route to host
E0222 10:04:28.041145       1 rest_v2.go:164] Unable to get cluster ID: Get "https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgmt/v2/cluster": dial tcp 172.30.136.191:443: connect: no route to host
E0222 10:04:28.041157       1 gpfs.go:222] Error getting cluster ID: Get "https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgmt/v2/cluster": dial tcp 172.30.136.191:443: connect: no route to host
E0222 10:04:28.041165       1 gpfs.go:164] Error in plugin initialization: Get "https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgmt/v2/cluster": dial tcp 172.30.136.191:443: connect: no route to host
F0222 10:04:28.041171       1 main.go:70] Failed to initialize Scale CSI Driver: Get "https://ibm-spectrum-scale-gui.ibm-spectrum-scale:443/scalemgmt/v2/cluster": dial tcp 172.30.136.191:443: connect: no route to host


```

Please let me know the work-around, For now this is blocking to make progress onto the Noobaa defect recreate 6869 

## Expected behavior

CSI pods running on worker0,worker2 (as worker1 is down), should be able to communicate with existing GUI pod , that way the noobaa-db-pg-0 that needs PV to get it to Running state would get the Cluster information for the DB to be coming into Running state for HPO operations to proceed. This is a blocker for the HPO solution if something like this happens in the field today, 

### Data Collection and Debugging

Environmental output

- What openshift/kubernetes version are you running, and the architecture? 
- `kubectl get pods -o wide -n < csi driver namespace> `
- `kubectl get nodes -o wide`
- CNSA/Spectrum Scale version 
- Remote Spectrum Scale version 
- Output for `./tools/spectrum-scale-driver-snap.sh -n < csi driver namespace> -v `

Tool to collect the CSI snap:

`./tools/spectrum-scale-driver-snap.sh -n < csi driver namespace>`

### Add labels

- Component:
- Severity:
- Customer Impact:
- Customer Probability:
- Phase:

Note : See [labels](https://github.com/IBM/ibm-spectrum-scale-csi/labels) for the labels

**Comment by deeghuge** [comment]
This is the defect in CNSA repo related to this https://github.ibm.com/IBMSpectrumScale/scale-core/issues/2951

**Comment by rkomandu** [comment]
@amdabhad , we are not denying that the communication would establish once the node is back. How long the node would be in the state is not a unknown from Customer POV. The problem is the HPO which has the noobaa-db-pg-0 is unable to communicate when the node is down, is going to stand-still all HPO operations . This is not going to work for the customers

**Comment by amdabhad** [comment]
@rkomandu Please ignore the last line in my earlier comment, it was just an additional observation I added, let me rather remove it to avoid confusion. The whole point or the summary of my last comment was that: there is no issue from CSI side, as I see this issue occurs without any involvement of CSI.
I am checking few more things, will add here if I find anything interesting.

**Comment by amdabhad** [comment]
An internal RedHat case created for this: https://access.redhat.com/support/cases/#/case/03163822

**Comment by dunnevan** [comment]
Just to add a note here: this is not directly related to Scale CSI itself.  This is in regard to CSI using the Scale cluster deployed via CNSA.  Please follow the internal CNSA issue 2951.

gist: the service for GUI endpoints is not removing the unready GUI pod (the one on the down node with the pod stuck terminating).  In this case, it seems to be some interaction of the service using SessionAffinity (set to 24 hours) and kube-proxy iptables backend not removing the unready, unserving, terminating pod's nat rules.  Since the nat rule has already been marked (by recent iptables module) it will continue to attempt to nat to the down pod.

**Comment by deeghuge** [comment]
Moving out of CSI 2.5.0 since it fixed via route as intermediate fix.

**Comment by Jainbrt** [comment]
@rkomandu could you please add Customer Impact & Customer probability labels to the issue?

**Comment by rkomandu** [comment]
@deeghuge , Are you planning to fix this defect in 2.5.1 ? as CNSA 2951 route is fixed in 513

**Comment by amdabhad** [comment]
There won't be any fix for this in CSI as it is fixed in CNSA - https://github.ibm.com/IBMSpectrumScale/scale-core/issues/2951
